docs/ref/request-response.txt

```diff
@@ -703,14 +703,18 @@ must use the :class:`StreamingHttpResponse` class instead.
 Setting header fields
 ~~~~~~~~~~~~~~~~~~~~~
 
-To set or remove a header field in your response, treat it like a dictionary::
+.. versionchanged:: 3.2
+
+    A new method of accessing headers was added. The old method of treating
+    the request object as a dictionary still works, but is undocumented.
```

Comment:
I think having some small documentation section would be good, so that it's discoverable/linkable for new developers. It could have a warning that it's not recommended and may be deprecated, like `index_together` and `unique_together`

Reply:
+1 to documenting the existing/old method. 

> ...It could have a warning that it's not recommended and may be deprecated... 

-1 on this. 

It shouldn't be deprecated. 1. That objects conform to the Python Data Model is good/great. 2. We'd be removing something that's been in place forever. There's no need to break code that's using it. (Django was 15 yesterday, perhaps by the time it's 30... i.e. never)

It's massively paternalistic to say what usage is recommended. It's Python. We're all consenting adults. 🙂

Reply:
> perhaps by the time it's 30... i.e. never

Are you suggesting Django won't make it to 30?? 😰

> It's massively paternalistic to say what usage is recommended.

I guess, but I think the reason we're adding `.headers` is because there's some consensus it's clearer. We should at least point out that `.headers` is newer.

Reply:
> Are you suggesting Django won't make it to 30?? 😰

My word, no! 🙂 More that we'd discuss removing `__setitem__` &co from responses at that time. 

No harm in advertising the new method — I think it will be more accessible to new users (& old too) — but I would strongly prefer that we don't imply that the dictionary access is somehow inferior, or under threat of removal. That I can treat a custom class, like `HttpResponse` in this case, as a standard Python object is one of the keys to the language's expressiveness. Let's just add the new property and leave well alone there. (IMO obviously)

Reply:
There is a bit more discussion on this on the list, so I'll refrain from making any changes until it's a bit more resolved.

Reply:
Positive suggestion: I'd put what you've got first, and then move the older example below it. 

```
You can also manipulate headers, treating your response like a dictionary::

...

This proxies to ``headers``, and is the original interface offered by ``HttpResponse``.

Note that unlike a dictionary, ``del`` doesn't raise ``KeyError`` if the header		
field doesn't exist.

.. versionchanged:: 3.2 (which should go at the bottom)
```
 
Not 100% sure about the "and is the original interface offered by ``HttpResponse``." but it maybe adds a touch of historical colour, and explains how come there are two ways.

Reply:
Carlton, this seems like a good compromise. I've added that almost verbatim.

Reply:
Thanks @knyghty. We can tweak but I'd be happy with something like this: it shows the old API one-time, but doesn't make a big thing of it, and highlights the newer approach first.